src/partition/repart.c

```diff
@@ -72,6 +72,17 @@
 /* LUKS2 takes off 16M of the partition size with its metadata by default */
 #define LUKS2_METADATA_SIZE (16*1024*1024)
 
+#if !HAVE_LIBCRYPTSETUP
+        struct crypt_device {
+
+        };
+        static inline void sym_crypt_free(struct crypt_device* cd) {
+            if (cd)
+                    free(cd);
+        }
+        DEFINE_TRIVIAL_CLEANUP_FUNC(struct crypt_device *, sym_crypt_free);
+#endif
```

Comment:
I find this definition of a fake structure very ugly. Normally we ifdef the relevant parts of various functions. In this case this will result in some quite visible ifdeffery, but I think that's still better than this approach.